lib/ably/PaginatedResource.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+class PaginatedResource extends ArrayObject {
+
+    private $ably;
+    private $domain;
+    private $path;
+    private $paginationHeaders = false;
+
+    /*
+     * Constructor
+     */
+    public function __construct( AblyRest $ably, $domain, $path, $params = array() ) {
+        parent::__construct();
+
+        $this->ably = $ably;
+        $this->domain = $domain;
+        $this->path = $path;
+
+        $withHeaders = true;
+        $response = $this->ably->get( $domain, $path, $this->ably->auth_headers(), $params, $withHeaders );
+
+        if (isset($response['body'])) {
+            $this->exchangeArray( $response['body'] );
```

Comment:
Should you check that $response['body'] is actually an array too? Also since we are on this subject, it would be nice to add some documentation around type returned and parameters.

I know it wasn't done yet, but since you created this class I think it's a good time to start :).